plugins/cache-caffeine/src/main/java/org/opensearch/cache/store/CaffeineHeapCache.java

```diff
@@ -83,26 +106,18 @@ private class CaffeineRemovalListener implements RemovalListener<ICacheKey<K>, V
         public void onRemoval(ICacheKey<K> key, V value, RemovalCause removalCause) {
             switch (removalCause) {
                 case SIZE:
-                    removalListener.onRemoval(
-                        new RemovalNotification<>(key, value, RemovalReason.CAPACITY)
-                    );
+                    removalListener.onRemoval(new RemovalNotification<>(key, value, RemovalReason.EVICTED));
                     cacheStatsHolder.incrementEvictions(key.dimensions);
                     break;
                 case EXPIRED:
-                    removalListener.onRemoval(
-                        new RemovalNotification<>(key, value, RemovalReason.INVALIDATED)
-                    );
+                    removalListener.onRemoval(new RemovalNotification<>(key, value, RemovalReason.INVALIDATED));
                     cacheStatsHolder.incrementEvictions(key.dimensions);
```

Comment:
I missed this in the first round, but evictions also shouldn't be incremented for the `EXPIRED` case

Reply:
@enarvade As mentioned by Peter, can u correct this?